pkg/lib/oauth/token_encoding.go

```diff
@@ -53,6 +61,30 @@ func (e *AccessTokenEncoding) EncodeAccessToken(client *config.OAuthClientConfig
 	// verified JWT.
 	_ = claims.Set(jwt.JwtIDKey, grant.TokenHash)
 
+	claimsMap, err := jwtutil.ToMap(claims)
+	if err != nil {
+		return "", err
+	}
+
+	eventPayload := &blocking.UserSessionJWTPreCreateBlockingEventPayload{
+		UserRef: model.UserRef{
+			Meta: model.Meta{
+				ID: userID,
+			},
+		},
+		Payload: claimsMap,
+	}
+
+	err = e.Events.DispatchEvent(eventPayload)
+	if err != nil {
+		return "", err
```

Comment:
When I return `is_allowed` to `false`, the SDK got the internal server error from the SDK. See if we can improve the error handling?